### DISCUSSION TITLE
errorHandler doesn't work anymore

### DISCUSSION THREAD

**Discussion by alexdieudonne** [payloadcms/payload#2245] [Q&A]
# Bug Report

Hello since the version 1.4.1 i was used to deal with the init.js with patch-package file to handle the errorHandle middleware inside payload, i had just to move 
-        payload.errorHandler = (0, errorHandler_1.default)(payload.config, payload.logger);
-        payload.router.use(payload.errorHandler);
from the bottom to the top now when i do that, i get a html render instead of my errorhandler function
version 1.6.1 after moving my things
<img width="890" alt="Screenshot 2023-03-05 at 11 57 39" src="https://user-images.githubusercontent.com/59718717/222956342-7d50b468-ff68-465f-9579-ac3b07f75f80.png">

version 1.5.2
<img width="858" alt="Screenshot 2023-03-05 at 11 53 58" src="https://user-images.githubusercontent.com/59718717/222956174-fa0f24bd-76b5-47e2-89fb-328b5960df46.png">

## Steps to Reproduce

1. Install version 1.5.2 payload 
2. move ` payload.errorHandler = (0, errorHandler_1.default)(payload.config, payload.logger) payload.router.use(payload.errorHandler);` from bottom to top of the function init or near top
3. Run and watch the difference between the 1.6.1 version

## Other Details
Obviously the errorHandler function didn't really works at expected.
This is my errorHandler
<img width="460" alt="Screenshot 2023-03-05 at 12 01 38" src="https://user-images.githubusercontent.com/59718717/222956530-9d0b8a37-8176-44d5-b545-f6e939a96499.png">

Thank you again.

**Comment by jmikrut** [comment]
Hey @alexdieudonne — I'm struggling to follow to be honest. You are reassigning `payload.errorHandler`? 

What is `ApiError`? And which `if` case is responsible for sending back the HTML as you've shown?

If I were you, I would return `res.json()` if you want the result of your error that is sent back to be properly formatted as JSON.

Our `errorHandler` file has not changed functionally in the last 18 months, so this will be tough to narrow down further on our end.

Internally, we just use `res.send()`, which should be responsible for automatically determining the `Content-Type` of the resultant error, but I am guessing that because you are only returning the message, it's treating your response as `text` instead of `json`.

I will convert this to a discussion unless we can identify exactly what the bug would be but we are happy to help further here. I would personally try and determine where your `payload.errorHandler` is actually sending the response, and try to structure your response as JSON. 

Here is where I'd check:

```
return res.status(err.status).send(err.message)
```

Try changing that to:

```
return res.status(err.status).send({ statusCode: err.status, message: err.message })`
```

Let me know if you can provide further info or if this fixes your issue!

**Reply by alexdieudonne** [reply]
### Thank you for your answer.



Okay i'll try to explain my situation more clearly
From the previous version 1.4.1 the `errorHandler function` payload didn't worked, so i tried by my own to figure out what's the problem and i've changed the line from my previous answer.

But since the newest version when i change 
`payload.errorHandler = (0, errorHandler_1.default)(payload.config, payload.logger) payload.router.use(payload.errorHandler);` and move this line to the top 
![Screenshot 2023-03-06 at 17 29 33](https://user-images.githubusercontent.com/59718717/223171472-2cc5e8a6-0ef8-425a-b800-af23d10b2a96.png)

I get that
<img width="887" alt="Screenshot 2023-03-06 at 17 30 18" src="https://user-images.githubusercontent.com/59718717/223171832-4ba680d5-e18e-4018-b29a-ae02e6c291a8.png">

The problem is with the previous version my errorHandler was called within the `init.js` but with the big changes, and implementation of classes inside the `initHTTP.js`, my errorHandler is not called anymore, but i have some doubt about the origin of that problem. (When i put a `console.log('test')` inside my function handler it doesn't log anything and cause of that i cannot send send a `res.json()`).

To answer to your questions

The ApiError is a class that i've created to parse the errors of payload and send custom ones with custom bad errors and that kind of stuff and there is no `if` to send html content, like i said earlier it works well with the previous version as i change the line where the errorHandler is called.
![Screenshot 2023-03-06 at 17 36 21](https://user-images.githubusercontent.com/59718717/223173341-30749df2-52dd-44f9-980b-ac7c7bae8b26.png)
